### ISSUE TITLE
Close Popups such as ColorPicker and Tooltips by code

### ISSUE THREAD

**Issue by cloa513** [godotengine/godot-proposals#4288]
### Describe the project you are working on

A Sudoku game

### Describe the problem or limitation you are having in your project

I can't allow the user to close the color choice by choosing a color. 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add a method to use the already existing code to close the popup. 

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

PopupState popup_state

Setter

set_popup_state(bool)
true opens the popup
false closes the popup 

Getter

get_popup_state()

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No equivalent that I can find- nearest is user manually closes the popup. Hide is not close. 

### Is there a reason why this should be core and not an add-on in the asset library?

Consistent treatment of nodes. You currently can't do a semiguided tour of a game where it starts with all tooltips open and when the user moves their mouse near any item with  tool tip then the tool tip closes.

**Comment by Hiiamwilliam** [comment]
Have you tried Control's `mouse_entered` and `mouse_exited`? It's clunky sometimes but for a ColorPicker I think it can work ok.

**Comment by cloa513** [comment]
> Have you tried Control's `mouse_entered` and `mouse_exited`? It's clunky sometimes but for a ColorPicker I think it can work ok.
Can you simulate those signals- send without actually moving the mouse?

**Comment by KoBeWi** [comment]
Something like this?
```GDScript
extends ColorPickerButton

func _ready() -> void:
	connect("color_changed", self, "hide_popup")

func hide_popup(c):
	get_popup().hide()
```

**Comment by cloa513** [comment]
No because hide is hide- the pop up is still there. The user could still click it if their mouse is over the popup area- any popup that are interactable except Colorpicker- I don't know? Also no idea how this interacts with the true close popup.

**Comment by KoBeWi** [comment]
No, hidden controls are not interactable. And `hide` is the method used to close popups.

**Comment by cloa513** [comment]
Some one might just want a hidden Colorpicker not closed. It is not well documented. Hide just says it sets Canvas.visible to false as in just about every version of Godot. Pop up also say nothing about how they close.

**Comment by KoBeWi** [comment]
Then it's a documentation issue.
